src/storage/src/source/mod.rs

```diff
@@ -466,7 +477,12 @@ pub enum NextMessage<Key, Value> {
 
 /// Source-agnostic wrapper for messages. Each source must implement a
 /// conversion to Message.
-pub struct SourceMessage<Key, Value> {
+#[derive(Clone, Serialize, Deserialize)]
```

Comment:
I might have missed it, where are we now serializing this struct?

Reply:
ahh -- hadn't thoroughly gone through and gotten rid of things that aren't necessary!  This was needed for v3 but I don't believe it's necessary for v4 -- same with a few `Data` bounds scattered around